src/__internal__/input/input.component.js

```diff
@@ -43,6 +44,12 @@ const Input = React.forwardRef(
       if (context.inputRef) context.inputRef(input);
     }, [context, input]);
 
+    useEffect(() => {
+      if (disabled) {
+        Browser.focusNextElement(input.current);
```

Comment:
I think we need to check whether the input is actually focused when it's disabled before focusing the next element, as we don't want to do this if it isn't

Reply:
As an example see this sandbox: https://codesandbox.io/s/carbon-quickstart-forked-pj15g?file=/src/index.js 

It's a simple Form where the third input is disabled. As you can see when the page loads, the focus immediately jumps to the fourth input because the third on is disabled